extension/src/popup/components/sendPayment/SendConfirm/index.tsx

```diff
@@ -0,0 +1,201 @@
+import React, { useReducer } from "react";
+import { useDispatch, useSelector } from "react-redux";
+
+import StellarSdk, { Horizon, Asset } from "stellar-sdk";
+import { Types } from "@stellar/wallet-sdk";
+
+import { navigateTo } from "popup/helpers/navigate";
+import { ROUTES } from "popup/constants/routes";
+import { signFreighterTransaction } from "popup/ducks/access";
+import { AppDispatch } from "popup/App";
+import { settingsNetworkDetailsSelector } from "popup/ducks/settings";
+
+import { PopupWrapper } from "popup/basics/PopupWrapper";
+import { BackButton } from "popup/basics/BackButton";
+
+import { SubmitFail, SubmitPending, SubmitSuccess } from "./SubmitResult";
+
+import "../styles.scss";
+
+const SUBMIT_STATE = {
+  idle: "idle",
+  pending: "pending",
+  success: "success",
+  fail: "failed",
+  view: "view",
+};
+enum ACTIONS {
+  submit = "submit",
+  success = "success",
+  fail = "fail",
+  view = "view",
+}
+const initialState = {
+  state: SUBMIT_STATE.idle,
+  response: null,
+  error: null,
+};
+
+const reducer = (
```

Comment:
@piyalbasu thoughts on this? I didn't think the submission state stuff should be in redux, but having two different dispatch methods in the same component does feel a bit odd. But too complicated for just a useState

Reply:
Hmmmm so I totally understand your apprehension in adding submission state to redux. It feels a bit weird to have UI state stored in memory, but it actually is often easier to store it there because we're often funneling async calls through redux. Since submission state is tied to the state of the async call, it's better to have the thing that monitors the async call (redux) also monitor the submission state. In fact, the redux lib we use in Freighter, redux-toolkit, standardizes this as an approach using built in pending/fulfilled/rejected states: https://redux-toolkit.js.org/api/createAsyncThunk#parameters

I would recommend creating a new `ducks` file called `internalTransaction` (or something along those lines), that has your existing `signFreighterTransaction` as well as a new `submitFreighterTransaction` action. You can use the pending/fulfilled/rejected cases to set submission state. Then, here in the UI, all you have to do is alter the UI based on changes in the Redux state. This new `duck` would be useful in the future when we want to sign/submit a different kind of transaction; we'd likely be able to reuse some of the logic here.

Take a look at https://github.com/stellar/freighter/blob/release/2.0.0/extension/src/popup/ducks/accountServices.ts and you can see an example of how to do this and also how something like `applicationState` can change based on the outcome of an async call. (We're not using `pending` in this file, but it's the same concept)

Reply:
Here's a simpler example of using redux toolkit to monitor async call status: https://github.com/stellar/account-viewer-v2/blob/master/src/ducks/account.ts

Reply:
👍 , changed to use redux lmk what you think
https://github.com/stellar/freighter/pull/386/commits/d0556addb5b71cac1acac38916bdf786713d156c